modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -58,7 +64,9 @@ ModularGapConductanceConstraint::ModularGapConductanceConstraint(const InputPara
     _adjusted_length(0.0),
     _disp_x_var(getVar("displacements", 0)),
     _disp_y_var(getVar("displacements", 1)),
-    _disp_z_var(_n_disp == 3 ? getVar("displacements", 2) : nullptr)
+    _disp_z_var(_n_disp == 3 ? getVar("displacements", 2) : nullptr),
+    _contact_pressure(coupledValueLower("contact_pressure")),
```

Comment:
There is the option of using `adCoupledLowerValue` with `ADVariableValue`, which should propagate the derivatives into the residuals through your user object, since you use an AD real. For aux kernels and other stuff pulling the regular version is fine, but here we could potentially benefit from a better Jacobian. I believe this'd be just a change of two lines.

Reply:
done 👍